### PR TITLE
Use Patch instead of Update in the reconciliation

### LINKED ISSUE #54: Change calls to Update() to Patch()
Instead of doing `Update()` calls for the changes we detect, we should attempt to use `Patch()`, to reduce the number of failures caused by stale objects.

From:
```go
		if err := params.Client.Update(ctx, updated); err != nil {
			return fmt.Errorf("failed to apply changes: %w", err)
		}
```

To:
```go
	// at the beginning of the reconciliation function, store a copy of `params.Instance` as `changed`
	patch := client.MergeFrom(&params.Instance)
	if err := params.Client.Patch(ctx, &changed, patch); err != nil {
		return fmt.Errorf("failed to apply changes: %w", err)
	}
```

### PR DESCRIPTION
**Pull request by treasurechristain**
Fixes #54 by changing the reconciliation calls to use Patch instead of Update.

### CODE REVIEW

**Comment by jpkrohling** [comment]
Looks like the code isn't compiling. Can you sort it out?

File: pkg/collector/reconcile/configmap.go [hunk 1]
```diff
@@ -117,10 +117,19 @@ func expectedConfigMaps(ctx context.Context, params Params, expected []corev1.Co
 			updated.ObjectMeta.Labels[k] = v
 		}
 
-		if err := params.Client.Update(ctx, updated); err != nil {
+		// editting Update with Patch
```
Comment (jpkrohling) [inline pkg/collector/reconcile/configmap.go:120]: You can remove this comment, the code is simple enough to be understood without it.

File: pkg/collector/reconcile/configmap.go [hunk 2]
```diff
@@ -117,10 +117,19 @@ func expectedConfigMaps(ctx context.Context, params Params, expected []corev1.Co
...
+		patch := client.MergeFrom(&params.Instance)
+
+		// if err := params.Client.Patch(ctx, &changed, patch); err != nil {
```
Comment (jpkrohling) [inline pkg/collector/reconcile/configmap.go:123]: You can also remove the commented out code. If we need it later, we can browse the git history.

**Review by jpkrohling — COMMENTED** [review]
Looks good! Just remove the comments, which aren't really necessary in this case, and take a look at the CLA check that is failing. I'll try it out manually soon and merge if everything works as expected!

**Comment by treasurechristain** [comment]
I just removed the comments I put in those files

**Review by jpkrohling — APPROVED** [review]
Looks good to me. Once you sort out the CLA check, it will get merged.

**Comment by treasurechristain** [comment]
I have checked the CLA. Hopefully it's merged now

**Pull request merged**